ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/offset/EarliestByOffset.java

```diff
@@ -123,4 +155,54 @@ public T map(final Struct agg) {
       }
     };
   }
+
+
+  @UdafFactory(description = "Earliest N by offset")
+  static <T> Udaf<T, List<Struct>, List<T>> earliestN(final Schema structSchema,
+      final int topNSize) {
+    return new Udaf<T, List<Struct>, List<T>>() {
```

Comment:
We'd want to validate `topNSize` (please add unit test for this too):

```suggestion
  static <T> Udaf<T, List<Struct>, List<T>> earliestN(
      final Schema structSchema,
      final int earliestN
    ) {
    
    if (earliestN <= 0) {
      throw new KsqlFunctionException("earliestN must be 1 or greater");
    }
    
    return new Udaf<T, List<Struct>, List<T>>() {
```

Reply:
Good point. Should there be an upper limit too?